pkg/agent/qrm-plugins/network/staticpolicy/util.go

```diff
@@ -0,0 +1,500 @@
+				filterOut = false
+			}
+		default:
+			filteredNICs = append(filteredNICs, nic)
+			filterOut = false
+		}
+
+		if filterOut {
+			general.Infof("filter out nic: %s mismatching with enhancement %s: %s",
+				nic.Name, consts.PodAnnotationNetworkEnhancementNamespaceType, consts.PodAnnotationNetworkEnhancementNamespaceTypeHost)
+		}
+	}
+
+	return filteredNICs
+}
+
+func filterNICsByHint(nics []*NetworkInterface, req *pluginapi.ResourceRequest, agentCtx *agent.GenericContext) []*NetworkInterface {
+	// means not to filter by hint (in topology hint calculation period)
+	if req.Hint == nil {
+		general.InfoS("req hint is nil, skip filterNICsByHint",
+			"podNamespace", req.PodNamespace,
+			"podName", req.PodName,
+			"containerName", req.ContainerName)
+		return nics
+	}
+
+	var exactlyMatchNIC *NetworkInterface
+	hintMatchedNICs := make([]*NetworkInterface, 0, len(nics))
+
+	hintNUMASet, err := machine.NewCPUSetUint64(req.Hint.Nodes...)
+
+	if err != nil {
+		general.Errorf("NewCPUSetUint64 failed with error: %v, filter out all nics", err)
+		return nil
+	}
+
+	for _, nic := range nics {
+		if nic == nil {
+			general.Warningf("found nil nic")
+			continue
+		}
+
+		siblingNUMAs, err := getSiblingNUMAs(nic.AffinitiveNUMANode, agentCtx.CPUTopology)
+
+		if err != nil {
+			general.Errorf("get siblingNUMAs for nic: %s failed with error: %v, filter out it",
+				nic.Name, err)
+			continue
+		}
+
+		if siblingNUMAs.Equals(hintNUMASet) {
+			// [TODO]: if multi-nics meets the hint, we need to choose best one according to left bandwidth or other properties
+			if exactlyMatchNIC == nil {
+				general.InfoS("add hint exactly matched nic",
+					"podNamespace", req.PodNamespace,
+					"podName", req.PodName,
+					"containerName", req.ContainerName,
+					"nic", nic.Name,
+					"siblingNUMAs", siblingNUMAs.String(),
+					"hintNUMASet", hintNUMASet.String())
+				exactlyMatchNIC = nic
+			}
+		} else if siblingNUMAs.IsSubsetOf(hintNUMASet) { // for pod affinity_restricted != true
+			general.InfoS("add hint matched nic",
+				"podNamespace", req.PodNamespace,
+				"podName", req.PodName,
+				"containerName", req.ContainerName,
+				"nic", nic.Name,
+				"siblingNUMAs", siblingNUMAs.String(),
+				"hintNUMASet", hintNUMASet.String())
+			hintMatchedNICs = append(hintMatchedNICs, nic)
+		}
+	}
+
+	if exactlyMatchNIC != nil {
+		return []*NetworkInterface{exactlyMatchNIC}
+	} else {
+		return hintMatchedNICs
+	}
+}
+
+func GetNetworkInterfaces(netNSDirAbsPath string) ([]*NetworkInterface, error) {
+	if netNSDirAbsPath == "" {
+		return nil, fmt.Errorf("GetNetworkInterfaces got nil netNSDirAbsPath")
+	}
+
+	var nics []*NetworkInterface
+
+	nsList := []string{""}
+	if dirs, err := ioutil.ReadDir(netNSDirAbsPath); err != nil {
+		return nil, err
+	} else {
+		for _, dir := range dirs {
+			if !dir.IsDir() {
+				nsList = append(nsList, dir.Name())
+			}
+		}
+	}
+
+	runtime.LockOSThread()
+	defer runtime.UnlockOSThread()
+	for _, ns := range nsList {
```

Comment:
I think we'd better filter out the "redundant" NICs here. Only those active host NICs are needed. For example, we can only allocate bandwidth on active physical NICs in a physical node.